File: 프로그래머스/0/120864. 숨어있는 숫자의 덧셈 （2）/숨어있는 숫자의 덧셈 （2）.py
def solution(my_string):
    answer = 0
    idx = 0
    num = ""
    N = len(my_string)
    
    while (idx < N):
        c = my_string[idx]
        if (not c.isnumeric()):
            if (num != ""):
                answer += int(num)
                num = ""
            idx += 1
            continue
        else:
            num += c
            idx += 1
    if num != "":
        answer += int(num)
    return answer

File: 프로그래머스/0/120864. 숨어있는 숫자의 덧셈 （2）/숨어있는 숫자의 덧셈 （2）.java
class Solution {
    public int solution(String my_string) {
        int answer = 0;
        int N = my_string.length();
        int idx = 0;
        String num = "";
        while (idx < N) {
            char c = my_string.charAt(idx);
            if (!Character.isDigit(c)) {
                if (num != "") {
                    answer += Integer.parseInt(num);
                }
                num = "";
                idx++;
                continue;
            }
            num += c;
            idx++;
        }
        if (num != "") {
            answer += Integer.parseInt(num);
        }
        return answer;
    }
}